server/text_generation_server/layers/attention/common.py

```diff
@@ -9,26 +9,48 @@
     @dataclass
     class Seqlen:
         input_lengths: torch.Tensor
+        prefix_lengths: torch.Tensor
         cu_seqlen_q: Optional[torch.Tensor]
         cu_seqlen_k: Optional[torch.Tensor]
+        max_q: int
+        max_k: int
 
-        def __init__(self, input_lengths):
+        def __init__(
+            self,
+            input_lengths,
+            prefix_lengths,
+            cu_seqlen_q=None,
+            max_q=None,
+            max_k=None,
+        ):
             self.input_lengths = input_lengths
+            self.prefix_lengths = prefix_lengths
             device = self.input_lengths.device
             shape = self.input_lengths.shape
-            cu_seqlen_q = torch.arange(
-                shape[0] + 1,
-                device=device,
-                dtype=torch.int32,
-            )
+            if cu_seqlen_q is None:
```

Comment:
When can it be None, and shouldn't it take into account speculative tokens into the arange?

Reply:
This is already done beforehand currently.

Althrough this should be fixed once speculative is correctly implemented with `cu_seqlen_q` and `cu_speculated_ids` or somethign (to get the offsets of the input_ids that are speculated for instance).